doc/source/whatsnew/v1.6.0.rst

```diff
@@ -243,7 +243,7 @@ Period
 
 Plotting
 ^^^^^^^^
--
+- Bug in :meth:`HistPlot._make_plot`, not dropping elements of ``weights`` corresponding to ``NaN`` values in ``data``.
```

Comment:
Could you write this in terms of the public api, `plot.hist`, as well as adding ```(:issue:`48884`)``` to the end?